C#-guidelines.md

```diff
@@ -0,0 +1,158 @@
+# C# Coding Guidelines
```

Comment:
This looks good, but are these written or sourced from somewhere?

If so, we should have a citation for where these came from.

Reply:
They are sourced and adjusted. Have included the references in the last section :)